File: specs/index_modules/index_settings/index_state.ts

/**namespace:IndexModules.IndexSettings */
/**custom_serialization*/
interface index_state {
	settings: map<string, any>[];
	aliases: map<index_name, alias>[];
	warmers: map<type_name, warmer>[];
	mappings: map<type_name, type_mapping>[];
	similarity: map<string, similarity>[];
}

File: output/csharp/IndexModules/IndexSettings/IndexState.cs
using Nest.CommonAbstractions;
using Nest.Internal;
using Nest.Indices;
using Nest.Mapping;
using System;
using System.Collections.Generic;
using System.Runtime.Serialization;
namespace Nest.IndexModules {

	public class IndexState  {
		
		[DataMember(Name="aliases")]
		public IDictionary<IndexName, Alias> Aliases { get; set; }


		[DataMember(Name="mappings")]
		public TypeMapping Mappings { get; set; }


		[DataMember(Name="settings")]
		public IDictionary<string, object> Settings { get; set; }

	}
}
